src/term/check/type_check.rs

```diff
@@ -66,6 +67,8 @@ fn unify(new: Type, old: &mut Type) -> Result<(), String> {
     (Type::Adt(new), Type::Adt(old)) if &new != old => {
       return Err(format!("Type mismatch. Found '{}' expected {}.", new, old));
     }
+    (Type::Tup, Type::Any) => *old = Type::Tup,
```

Comment:
Can you make it slightly simpler by changing these cases to just `(new, Type::Any) => *old = new`?